app/controllers/course/forum/topics_controller.rb

```diff
@@ -0,0 +1,116 @@
+class Course::Forum::TopicsController < Course::ComponentController
```

Comment:
follow the rest of the controllers to inherit from a Forum::Controller that loads the forum.